nixos/modules/services/databases/postgresql.nix

```diff
@@ -23,6 +23,11 @@ let
       ${cfg.extraConfig}
     ''; 
 
+  configFileCheck = pkgs.runCommand "postgresql-configfile-check" {} ''
+    ${cfg.package}/bin/postgres -D${configFile} -C config_file >/dev/null
```

Comment:
See also #62853 for cross-compile support.

Reply:
this approach does not work so well for postgresql because the right version of the package should be used...